### DISCUSSION TITLE
wont start - fails to find socket of source postgres

### DISCUSSION THREAD

**Discussion by ITJamie** [tianon/docker-postgres-upgrade#66] [General]
```
docker run --rm \
    -v /srv/appname/data/postgres:/var/lib/postgresql/10 \
    -v /srv/appname/data/postgres12:/var/lib/postgresql/12 \
    tianon/postgres-upgrade:10-to-12 

Performing Consistency Checks
-----------------------------
Checking cluster versions                                   ok

*failure*
Consult the last few lines of "pg_upgrade_server.log" for
the probable cause of the failure.

connection to database failed: connection to server on socket "/var/lib/postgresql/.s.PGSQL.50432" failed: No such file or directory
	Is the server running locally and accepting connections on that socket?

could not connect to source postmaster started with the command:
"/usr/lib/postgresql/10/bin/pg_ctl" -w -l "pg_upgrade_server.log" -D "/var/lib/postgresql/10/data" -o "-p 50432 -b  -c listen_addresses='' -c unix_socket_permissions=0700 -c unix_socket_directories='/var/lib/postgresql'" start
Failure, exiting
```

Any ideas of next troubleshooting step?
Ive confirmed all the files in /srv/appname/data/postgres are owned by uid 1001. ive tried the 10-14 image and it was the same error
Ive also tried adding "data" to the volume mounts but the same issue occurs, if i put in an incorrect path it warns that it cant find the PG_VERSION file, so its definitely seeing the data:
```
docker run --rm \
    -v /srv/appname/data/postgres/data:/var/lib/postgresql/10/data \
    -v /srv/appname/data/postgres12/data:/var/lib/postgresql/12/data \
    tianon/postgres-upgrade:10-to-12 
```

**Comment by ITJamie** [comment]
just a note. the source postgres container data dir was a https://hub.docker.com/r/bitnami/postgresql container

**Comment by ITJamie** [comment]
so it turns out the bitnami data folder doesnt contain a postgres.conf file or a pg_hba file. they move those elsewhere and are customised. sigh

**Reply by cheretbe** [reply]
@ITJamie Did you solve this problem? My case is even worse: custom-built local image based on postgres-alpine with manually installed patroni controlling postgres. 
There are `postgresql.conf` and `pg_hba.conf` in the data dir, but pg_ctl call fails.

**Reply by ITJamie** [reply]
I moved to postgres's own container

**Reply by cheretbe** [reply]
Most likely I'm going to do the same. Will require more work than just a version upgrade though...
Thanks.
